### ISSUE TITLE
Dev environment zcashd dies on restart, predictably

### ISSUE THREAD

**Issue by holmesworcester** [ZbayApp/zbay#279]
In my current setup, running `npm run mainnet` starts zcashd, validates a few hundred transactions, and dies with the following error message:

```
znode-mainnet_1  | zcashd: wallet/wallet.cpp:1080: void CopyPreviousWitnesses(NoteDataMap&, int, int64_t) [with NoteDataMap = std::map<SaplingOutPoint, SaplingNoteData>; int64_t = long int]: Assertion `(nd->witnessHeight == -1) || (nd->witnessHeight == indexHeight - 1)' failed.
```

@jkdev601 is there a zcash issue for this? What are the steps to get it working again?

**Comment by NorbertBodziony** [comment]
Zcash issue https://github.com/zcash/zcash/issues/4301. To solve this you need to recreate docker container. I usually just change command in `docker-compose.yml` file just remove `-debug` flag or you need to add `--force-recreate` flag to `docker-compose up znode-mainnet` command.

**Comment by holmesworcester** [comment]
Okay, I just removed `-debug` flag in `docker-compose.yml`

**Comment by holmesworcester** [comment]
The above caused zcashd to redownload a bunch of blocks. However I still got an error: `zbay_znode-mainnet_1 exited with code 137` @NorbertBodziony .

I tried `docker-compose up  --force-recreate znode-mainnet` and i'm waiting.

**Comment by holmesworcester** [comment]
That also dies with `zbay_znode-mainnet_1 exited with code 137`

**Comment by NorbertBodziony** [comment]
Yes since all data is embedded inside image recreation forces to redownload and rescan all data.

**Comment by holmesworcester** [comment]
@NorbertBodziony what should i do to get my dev environment working again?

**Comment by NorbertBodziony** [comment]
Old version is deprecated right now @jkdev601 is already working on building new one I will let you know when its up. But when that situation happens again you again solutions that I provided to you should be fine.
